src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSignedCommits.java

```diff
@@ -0,0 +1,162 @@
+package com.sap.sgs.phosphor.fosstars.data.github;
+
+import static com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures.USES_VERIFIED_SIGNED_COMMITS;
+
+import com.sap.sgs.phosphor.fosstars.data.UserCallback;
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.BooleanValue;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoSkipQuestion;
+import com.sap.sgs.phosphor.fosstars.tool.YesNoSkipQuestion.Answer;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * This data provider tries to figure out if a project uses signed commits. First, it checks if a
+ * project belongs to an organization and checks if any commits are signed or verified. It gathers
+ * the list of commits within one year from the current date. Then it calculates density ratio
+ * between number of signed commits vs unsigned commits. If the density ratio >= 1, then we set the
+ * flag = true else flag = false. Foundation. Next, it tries to ask a user if {@link UserCallback}
+ * is available.
+ */
+public class UsesSignedCommits extends AbstractGitHubDataProvider {
+
+  /**
+   * 1 year in millis.
+   */
+  private static final long DELTA = 365 * 24 * 60 * 60 * 1000L;
+
+  /**
+   * The percentage threshold of number of signed commits against number of total commits.
+   */
+  private static final double SIGNED_COMMIT_PRECENTAGE_THRESHOLD = 0.4;
+
+  /**
+   * A git token provided by user to access the Github API.
```

Comment:
Please change it to "A GitHub token ..." since git doesn't have tokens but GitHub does. Also, please use "GitHub" (capital H). Sorry :) Then, it would be better to rename `gitToen` to `githubToken`.